File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.cpp
class Solution {
public:
    int strStr(string s, string needle) {
        
        string sample = "";

        if (needle.length() > s.length()) return -1;

        for (int i=0; i<needle.size(); i++) {
            sample += s[i];
        }

        if (sample == needle) return 0;
        int l=0 , r = sample.length()-1;
        while (r < s.length()) {

            sample.erase(0 , 1);
            l++;
            r++;
            if (r == s.length()) break;
            sample += s[r];

            if (sample == needle) return l;



        }

        return -1;

    }
};

File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.py
class Solution:
    def strStr(self, s: str, needle: str) -> int:

        sample = ""

        if len(needle) > len(s):
            return -1

        for i in range(len(needle)):
            sample += s[i]

        if sample == needle:
            return 0

        l = 0
        r = len(sample) - 1

        while r < len(s):

            sample = sample[1:]
            l += 1
            r += 1

            if r == len(s):
                break

            sample += s[r]

            if sample == needle:
                return l

        return -1